### PR TITLE
Adopt Hono middleware convention and support Remix v2

### PR DESCRIPTION
**Pull request by rphlmr**
Hey 👋

This PR aims to follow Hono convention for writing custom middleware. 
It uses [createMiddleware](https://hono.dev/helpers/factory#createmiddleware) to reduce the type burden. 
I've also added some tests.

It should solve #53 and I hope it will help adopt Remix Vite (not yet tested).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,40 +64,40 @@
...
+		"@remix-run/server-runtime": "^2.5.0",
+		"hono": "^3.12.5",
 		"pretty-cache-header": "^1.0.0"
 	},
 	"optionalDependencies": {
-		"@remix-run/cloudflare": "^1.19.1",
-		"i18next": "^23.4.1",
-		"remix-i18next": "^5.3.0",
-		"zod": "^3.21.4"
+		"@remix-run/cloudflare": "^2.5.0",
```
Comment (sergiodxa) [inline package.json:72]:
```suggestion
		"@remix-run/cloudflare": "^2.0.0",
```

File: package.json [hunk 2]
```diff
@@ -64,40 +64,40 @@
...
+		"i18next": "^23.7.16",
```
Comment (sergiodxa) [inline package.json:73]:
```suggestion
		"i18next": "^23.0.0",
```

File: package.json [hunk 3]
```diff
@@ -64,40 +64,40 @@
...
+		"remix-i18next": "^5.5.0",
```
Comment (sergiodxa) [inline package.json:74]:
```suggestion
		"remix-i18next": "^5.0.0",
```

File: package.json [hunk 4]
```diff
@@ -64,40 +64,40 @@
...
+		"zod": "^3.22.4"
```
Comment (sergiodxa) [inline package.json:75]:
```suggestion
		"zod": "^3.0.0"
```

File: src/security.ts [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
+
+const https = "https:";
 
 /**
  * Enforce HTTPS connections.
  */
-export function httpsOnly(): MiddlewareHandler {
-	return async function middleware(context, next) {
-		let url = new URL(context.req.url);
-		if (url.protocol !== "http:") return await next();
```
Comment (sergiodxa) [inline src/security.ts:9]: This check if it's not http: so other protocols also pass, only http: should be redirected to https:

**Review by sergiodxa — APPROVED** [review]

File: src/security.ts (hunk 5 above)
Comment (rphlmr) [reply src/security.ts:9]: Ah, yes, my bad!

Comment (rphlmr) [reply src/security.ts:9]: I let you resolve this one if it's good for you?
Thanks for the review.

**Review by sergiodxa — APPROVED** [review]

**Pull request merged**